### PR TITLE
Add support for RSS on linux

### PR DESCRIPTION
**Pull request by ThadHouse**
We had a thread per socket context, so we only needed to configure the bpf for rss to get it working

### CODE REVIEW

File: src/perf/lib/RpsClient.cpp [hunk 1]
```diff
@@ -397,7 +397,9 @@ RpsConnectionContext::StreamCallback(
     case QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE:
         Worker->Client->StreamContextAllocator.Free(StrmContext);
         MsQuic->StreamClose(StreamHandle);
-        Worker->QueueSendRequest();
+        if (CxPlatTimeMs32() < (Worker->Client->RunTime + 3000)) {
```
Comment (nibanks) [inline src/perf/lib/RpsClient.cpp]: This is wrong. There should be a diff between times here. Also 3000 ms is too much IMO. 100 ms might be enough.

File: src/perf/lib/RpsClient.cpp [hunk 2]
```diff
@@ -397,7 +399,10 @@ RpsConnectionContext::StreamCallback(
     case QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE:
         Worker->Client->StreamContextAllocator.Free(StrmContext);
         MsQuic->StreamClose(StreamHandle);
-        Worker->QueueSendRequest();
+        if ((CxPlatTimeMs32() - Worker->Client->StartTime)
```
Comment (nibanks) [inline src/perf/lib/RpsClient.cpp:402]: Use `CxPlatTimeDiff32` instead.

File: scripts/performance-helper.psm1 [hunk 3]
```diff
@@ -224,8 +224,15 @@ function Wait-ForRemote {
     param ($Job)
     # Ping sidechannel socket on 9999 to tell the app to die
     $Socket = New-Object System.Net.Sockets.UDPClient
-    $Socket.Send(@(1), 1, $RemoteAddress, 9999) | Out-Null
-    Wait-Job -Job $Job -Timeout 120 | Out-Null
+    for ($i = 0; $i -lt 120; $i++) {
+        $Socket.Send(@(1), 1, $RemoteAddress, 9999) | Out-Null
+        $Socket.Send(@(1), 1, $RemoteAddress, 9999) | Out-Null
```
Comment (nibanks) [inline scripts/performance-helper.psm1:229]: Why two? Seems unnecessary.

File: src/perf/lib/RpsClient.cpp [hunk 4]
```diff
@@ -397,7 +399,10 @@ RpsConnectionContext::StreamCallback(
     case QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE:
         Worker->Client->StreamContextAllocator.Free(StrmContext);
         MsQuic->StreamClose(StreamHandle);
-        Worker->QueueSendRequest();
+        if (CxPlatTimeDiff32(Worker->Client->StartTime, CxPlatTimeMs32()) 
+            < (Worker->Client->RunTime + 3000)) {
```
Comment (nibanks) [inline src/perf/lib/RpsClient.cpp:403]: Still think `3000` is too much. No more than `1000` but I think `100` is also good.

File: src/platform/platform_linux.c [hunk 5]
```diff
@@ -734,6 +734,27 @@ CxPlatThreadCreate(
...
+    CPU_ZERO(&CpuSet);
+    CPU_SET(ProcessorIndex, &CpuSet);
+
+    if (!pthread_setaffinity_np(Thread, sizeof(CpuSet), &CpuSet)) {
+        QuicTraceEvent(
+            LibraryError,
+            "[ lib] ERROR, %s.",
+            "pthread_setaffinity_np failed");
+    }
+    return QUIC_STATUS_SUCCESS;
```
Comment (nibanks) [inline src/platform/platform_linux.c:755]: nit: newline before this

File: src/platform/platform_linux.c [hunk 6]
```diff
@@ -734,6 +734,27 @@ CxPlatThreadCreate(
...
 
+QUIC_STATUS
+CxPlatSetCurrentThreadProcessorAffinity(
+    _In_ uint16_t ProcessorIndex
+    )
+{
+    pthread_t Thread;
+    cpu_set_t CpuSet;
+
+    Thread = pthread_self();
```
Comment (nibanks) [inline src/platform/platform_linux.c:745]: nit:
```suggestion
    cpu_set_t CpuSet;
    pthread_t Thread = pthread_self();
```

File: src/platform/datapath_linux.c [hunk 7]
```diff
@@ -1616,6 +1658,11 @@ CxPlatSocketCreateUdp(
         }
     }
 
+    Status = CxPlatSocketConfigureRss(&Binding->SocketContexts[0], SocketCount);
+    if (QUIC_FAILED(Status)) {
+        goto Exit;
+    }
```
Comment (nibanks) [inline src/platform/datapath_linux.c:1664]: Is this universally supported? What happens if it's not?

File: scripts/performance-helper.psm1 (hunk 3 above)
Comment (ThadHouse) [reply scripts/performance-helper.psm1:229]: I thought it was failing on the remote side, not the client side, so I just wanted to make sure they would send. I'll bring it back down to 1 now.

**Review by nibanks — APPROVED** [review]

File: src/platform/datapath_linux.c (hunk 7 above)
Comment (ThadHouse) [reply src/platform/datapath_linux.c]: Probably a good point to not make this fail. Removed the status check, and added a note on why we're ignoring the status.

**Pull request merged**